draft-ietf-core-coap-tcp-tls.md

```diff
@@ -112,36 +112,48 @@ Where NATs are still present,
 CoAP over TCP can also help with their traversal. NATs often calculate
 expiration timers
 based on the transport layer protocol being used by application protocols.
-Many NATs are built around the assumption that a transport layer protocol
+Many NATs are built around the assumption that a transport layer protocol,
 such as
-TCP gives them additional information about the session life cycle
+TCP, gives them additional information about the session life cycle
 and keep TCP-based NAT bindings around for a longer period. UDP, on the other
 hand,
 does not provide such information to a NAT and timeouts tend to be
 much shorter, as research confirms {{HomeGateway}}.
 
-Some environments may also benefit from the more sophisticated congestion
-control capabilities provided by many TCP implementations.
+Some environments may also benefit from the ability of TCP to exchange
+larger payloads (such as firmware images) without application layer
+segmentation and to utilize the more
+sophisticated congestion control capabilities provided by many TCP implementations.
 (Note that there is ongoing work to add more elaborate congestion control
 to CoAP as well, see {{-cocoa}}.)
 
 Finally, CoAP may be integrated into a Web environment where the front-end
 uses CoAP from IoT devices to a cloud infrastructure but the CoAP messages
 are then transported in TCP between the back-end services.
-A TCP-to-UDP gateway can be
-used at the cloud boundary to talk to the UDP-based IoT.
+A TCP-to-UDP gateway can be used at the cloud boundary to talk to the UDP-based IoT.
 
 To make IoT devices work smoothly in these demanding environments, CoAP
 needs
 to make use of a different transport protocol, namely TCP {{RFC0793}},
 in some situations secured by TLS {{RFC5246}}.
 
-The present document
-describes a shim header that conveys length information about
-each CoAP message.  Modifications to CoAP beyond the replacement
-of the message layer
-(e.g., to introduce further optimizations) are intentionally avoided.
+Conceptually, the CoAP over TCP/TLS specification replaces most of
```

Comment:
should not block the merge, but we should rework that paragraph as it a very heavy sentence